R/dataframe__frame.R

```diff
@@ -1682,6 +1682,74 @@ DataFrame_sample = function(
 }
 
 
+#' Transpose a DataFrame over the diagonal.
+#'
+#' @param include_header boolean (default FALSE) if TRUE, add previous column names as
+#' a new Utf8 column.
+#' @param header_name string name of the header column containing previous column names. Only
+#' relevant if include_header = TRUE.
+#' @param column_names character vector or default NULL. How to name previous rows as columns.
+#' NULL is default and is "column_1", "column_2" ... . Length of column names must match n rows of
+#' input DataFrame.
+#'
+#' @details
+#' This is a very expensive operation. Perhaps you can do it differently.
+#'
+#' If you need to perform non foldable (see fold / reduce) row operations like median.
+#' Transpose may be the fastest option, for that.
+#'
+#' Polars transpose is currently eager only, likely because it is not trivial to deduce the schema.
+#'
+#' @keywords DataFrame
+#' @return DataFrame
+#' @examples
+#'
+#' # simple use-case
+#' pl$DataFrame(mtcars)$transpose(include_header = TRUE, column_names = rownames(mtcars))
+#'
+#' # All rows must have one shared supertype, recast Categorical to Utf8 which is a supertype
+#' # of f64, and then dataset "Iris" can be transposed
+#' pl$DataFrame(iris)$with_columns(pl$col("Species")$cast(pl$Utf8))$transpose()
+#'
+#' # An example where transpose is warranted for speed.
```

Comment:
I feel like the whole example below is useful but shouldn't be placed here. It shows a lot of things (chunks, fold, reduce, etc.) that are not related to `$transpose()`. 

Maybe there should be a kind of vignettes like "Useful tips and tricks with polars" where we can gather this kind of examples

Reply:
You are probably right. I remove the example.